.markdownlint.json

```diff
@@ -0,0 +1,9 @@
+{
+    "default": false,
```

Comment:
What does setting it to false do? Does it disable other rules?

Reply:
The definition says: "// Default state for all rules" in https://github.com/DavidAnson/markdownlint/blob/main/schema/.markdownlint.jsonc

I guess it prevents rules that are enabled by default rules.

Reply:
So, why do you set it to false?

Reply:
I don't want to enable all the rules implicitly but enable only line length checker for now. So I think when I set it to false, it does the job.

Just tried, if I set it to true, some additional warnings generated

Reply:
How many of them? Maybe we can just include those fixes as well

Reply:
```
code_samples/jaas_sample/hazelcast-member/README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "Make sure you have"]
code_samples/jaas_sample/hazelcast-member/README.md:10 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
code_samples/jaas_sample/hazelcast-member/README.md:16 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
code_samples/jaas_sample/README.md:28 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
code_samples/jaas_sample/README.md:38 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
CONTRIBUTING.md:34:32 MD033/no-inline-html Inline HTML [Element: abbr]
DOCUMENTATION.md:126 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Introduction"]
DOCUMENTATION.md:132 MD001/heading-increment/header-increment Heading levels should only increment by one level at a time [Expected: h2; Actual: h3]
DOCUMENTATION.md:144 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 1. Getting Started"]
DOCUMENTATION.md:196 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:248 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:424 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:456 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Using a Map"]
DOCUMENTATION.md:462 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "it.js"]
DOCUMENTATION.md:479 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Output"]
DOCUMENTATION.md:481 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:492 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Sales.js"]
DOCUMENTATION.md:509 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Output"]
DOCUMENTATION.md:511 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:531 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 2. Features"]
DOCUMENTATION.md:582 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 3. Configuration"]
DOCUMENTATION.md:604 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 3.1 Client Properties"]
DOCUMENTATION.md:627 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 4. Serialization"]
DOCUMENTATION.md:649 MD028/no-blanks-blockquote Blank line inside blockquote
DOCUMENTATION.md:655:1 MD038/no-space-in-code Spaces inside code span elements [Context: "`"]
DOCUMENTATION.md:657 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Serialization Priority"]
DOCUMENTATION.md:1043 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 5. Setting Up Client Network"]
DOCUMENTATION.md:1202 MD028/no-blanks-blockquote Blank line inside blockquote
DOCUMENTATION.md:1206 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 6. Client Connection Strateg..."]
DOCUMENTATION.md:1289 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 7. Blue-Green Deployment and..."]
DOCUMENTATION.md:1352 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 8. Using Node.js Client with..."]
DOCUMENTATION.md:2155 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:2301 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Entry Listener Functions Suppo..."]
DOCUMENTATION.md:2307 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Entry Listener Functions Suppo..."]
DOCUMENTATION.md:2537 MD028/no-blanks-blockquote Blank line inside blockquote
DOCUMENTATION.md:2789 MD036/no-emphasis-as-heading/no-emphasis-as-header Emphasis used instead of a heading [Context: "Predicates Object Operators"]
DOCUMENTATION.md:2849:115 MD033/no-inline-html Inline HTML [Element: version]
DOCUMENTATION.md:3584 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 9. Securing Client Connectio..."]
DOCUMENTATION.md:3779 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 10. Development and Testing"]
DOCUMENTATION.md:3798 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:3805 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:3812 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:3825 MD040/fenced-code-language Fenced code blocks should have a language specified [Context: "```"]
DOCUMENTATION.md:3831 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 11. Getting Help"]
DOCUMENTATION.md:3840 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 12. Contributing"]
DOCUMENTATION.md:3845 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 13. License"]
DOCUMENTATION.md:3849 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# 14. Copyright"]
README.md:1:1 MD033/no-inline-html Inline HTML [Element: p]
README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "<p align="center">"]
README.md:2:5 MD033/no-inline-html Inline HTML [Element: a]
README.md:3:9 MD033/no-inline-html Inline HTML [Element: img]
README.md:5:5 MD033/no-inline-html Inline HTML [Element: h2]
README.md:8:1 MD033/no-inline-html Inline HTML [Element: p]
README.md:9:5 MD033/no-inline-html Inline HTML [Element: a]
README.md:10:9 MD033/no-inline-html Inline HTML [Element: img]
README.md:12:5 MD033/no-inline-html Inline HTML [Element: a]
README.md:13:9 MD033/no-inline-html Inline HTML [Element: img]
README.md:15:5 MD033/no-inline-html Inline HTML [Element: a]
README.md:17:9 MD033/no-inline-html Inline HTML [Element: img]
test/integration/backward_compatible/README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "This folder contains integrati..."]
test/soak_test/README.md:16 MD025/single-title/single-h1 Multiple top-level headings in the same document [Context: "# Success Criteria"]
test/unit/statistics/README.md:1 MD041/first-line-heading/first-line-h1 First line in a file should be a top-level heading [Context: "The metrics binary compatibili..."]
```

These are the failing ones. Still I don't want to include defaults implicitly, just to be explicit what we are checking. If you want any of rules to be included, you can suggest.

Reply:
MD040
MD001

The rules above look useful and easy to fix to me.

Also, we probably should use `##` for this heading

[Context: "# 3.1 Client Properties"]

Reply:
Thanks fixed them and also changed scripts structure a little bit in  https://github.com/hazelcast/hazelcast-nodejs-client/pull/1040/commits/827273c3c92f163f6d94ea46c3ed75d0f897b6e1

Reply:
Thanks for the changes.

I couldn't see the fix for the following part of my comment above. Can you include the fix for it as well?

Also, we probably should use ## for this heading

[Context: "# 3.1 Client Properties"]

Reply:
Oh I changed that part but it is somehow reverted, sorry. Sending the change now.

FYI, I will change scripts a little bit. `lint-markdown` script has auto fix flag that auto fixes files for certain rules. I will add a `lint-markdown:fix` script like we did with eslint. Then only `lint:fix` script will use `lint-markdown:fix`, lint will use `lint-markdown` that won't autofix. Currently both `lint` and `lint:fix` auto fix errors in markdown files since -f flag is present. I think scripts that are changing files should be more clear, so that's why I chose this approach.